File: code-java/zipkin-service/src/main/java/com/example/zipkinservice/ZipkinServiceApplication.java
package com.example.zipkinservice;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
import zipkin.server.EnableZipkinServer;

@EnableDiscoveryClient
@EnableZipkinServer
@SpringBootApplication
public class ZipkinServiceApplication {

    public static void main(String[] args) {
        SpringApplication.run(ZipkinServiceApplication.class, args);
    }
}

File: code-kotlin/zipkin-service/src/main/kotlin/com/example/zipkinservice/ZipkinServiceApplication.kt
package com.example.zipkinservice

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.cloud.client.discovery.EnableDiscoveryClient
import zipkin.server.EnableZipkinServer

@EnableDiscoveryClient
@EnableZipkinServer
@SpringBootApplication
class ZipkinServiceApplication

fun main(args: Array<String>) {
    SpringApplication.run(ZipkinServiceApplication::class.java, *args)
}
